crates/pink-drivers/log_server/lib.rs

```diff
@@ -10,14 +10,33 @@ mod contract {
     #[ink(storage)]
     pub struct Contract {}
 
+    fn start_sidevm() {
+        let code_hash = *include_bytes!("./sideprog.wasm.hash");
+        pink::start_sidevm(code_hash).expect("Failed to start sidevm");
+    }
+
     impl Contract {
         #[ink(constructor)]
         pub fn default() -> Self {
-            let code_hash = *include_bytes!("./sideprog.wasm.hash");
-            pink::start_sidevm(code_hash).expect("Failed to start sidevm");
+            start_sidevm();
             Self {}
         }
 
+        #[ink(message)]
+        pub fn version(&self) -> this_crate::VersionTuple {
+            this_crate::version_tuple!()
+        }
+
+        #[ink(message)]
+        pub fn start(&self) {
```

Comment:
Anyone can call?

Reply:
Added permission check, and added a `#[ink(message)] fn owner()` for each driver.

Reply:
Is owner the system contract or the developer who deployed it?

Reply:
The one who deployed it. The system contract already has permission to stop any contract's sidevm.